tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion.h

```diff
@@ -108,6 +113,8 @@ class GpuHorizontalLoopFusion : public HloModulePass {
 
  private:
   StatusOr<bool> RunOnComputation(HloComputation*);
+  std::string prefix_;
+  bool skip_if_control_flow_;
```

Comment:
Note, in theory this flag isn't needed.
I added it to make sure I do not cause any regression.
I think we should try to remove it in another PR a week or so after this one is merged.

I do not know if control-flow can be present in the first run of the optimization or not.
If it can be present, this is a potential issue. But I didn't hear report of this, so removing completely the pass would slow done existing cases.
If you are sure that control-flow can't be present before the first pass of this optimization, I can remove it now.
Otherwise, let remove it a little bit later. This will help clarify the issue and make the gain available earlier.

Reply:
Before just doing more tests, I have a question.  Where control flow can be introduced in the XLA graph? Is it only in the copy_insertion optimization pass?

Reply:
I'm not 100% sure, but actually I can't think of any passes adding control flow nodes (if- and while-). We do have some "linearization" passes adding direct dependencies between HLO instructions.

Reply:
Ok. So can have control flow inserted before the fusions.
So mostly, we need to fix the fusion optimizations to handle this.
I just pushed a commit that remove that variable.